resourcename/join.go

```diff
@@ -0,0 +1,56 @@
+package resourcename
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Join combines resource names, separating them by slashes.
+func Join(elems ...string) (string, error) {
```

Comment:
Might be nitpicking here but do we want Join to throw an error? Most Join methods don't - and if we use the scanner there should be an implementation where we simply scan all inputs and append them to each other